tuner/libtuner.py

```
@@ -0,0 +1,1302 @@
+    ) -> list[str]:
+        pass
+
+    @abstractmethod
+    def get_dispatch_benchmark_command(
+        self, candidate_tracker: CandidateTracker
+    ) -> list[str]:
+        pass
+
+    @abstractmethod
+    def get_model_compile_command(
+        self, candidate_tracker: CandidateTracker
+    ) -> list[str]:
+        pass
+
+    @abstractmethod
+    def get_model_benchmark_command(
+        self, candidate_tracker: CandidateTracker
+    ) -> list[str]:
+        pass
+
+
+@dataclass
+class TaskPack:
+    args: argparse.Namespace
+    candidate_id: int
+    command: list[str]
+    check: bool = True
+    command_need_device_id: bool = False
+    cooling_time: int = 0
+
+
+@dataclass
+class TaskResult:
+    result: subprocess.CompletedProcess
+    candidate_id: int
+    device_id: str
+
+
+@dataclass
+class ParsedDisptachBenchmarkResult:
+    candidate_id: int
+    benchmark_time_in_seconds: float
+    candidate_mlir: Path
+    candidate_spec_mlir: Path
+
+
+@dataclass
+class IREEBenchmarkResult:
+    # Default format follows output of iree-benchmark-module
+    candidate_id: int
+    result_str: str
+
+    def get_mean_time(self) -> Optional[float]:
+        if not self.result_str:
+            return None
+        pattern = r"process_time/real_time_mean\s+([\d.]+)\s\w{2}"
+        match = re.search(pattern, self.result_str)
+        if not match:
+            return None
+        try:
+            return float(match.group(1))
+        except ValueError:
+            return None
+
+
+def generate_display_DBR(
+    candidate_id: int = 0, mean_time: float = random.uniform(100.0, 500.0)
+) -> str:
+    """Generate dispatch_benchmark_result string for displaying"""
+    return f"{candidate_id}\tMean Time: {mean_time:.1f}"
+
+
+def generate_display_MBR(
+    candidate_vmfb_path_str: str = "baseline.vmfb",
+    device_id: str = "0",
+    t1: float = random.uniform(100.0, 500.0),
+    calibrated_diff: Optional[float] = None,
+) -> str:
```

Same comment about defaulting to a random value.